TLM/TLM/Manager/Impl/LaneArrowManager.cs

```diff
@@ -297,20 +364,13 @@ public static class SeparateTurningLanes{
                         true
                         );
                 int srcLaneCount = laneList.Count();
-                for(int i = 0; i < srcLaneCount; ++i) {
-                    if(LaneConnectionManager.Instance.HasConnections(laneList[i].laneId, startNode)) {
-                        res = SetLaneArrowError.LaneConnection;
-                        return;
-                    }
-                }
-
                 if (srcLaneCount <= 1) {
                     return;
                 }
 
-                int leftLanesCount = CountTargetLanesTowardDirection(segmentId, nodeId, ArrowDirection.Left);
-                int rightLanesCount = CountTargetLanesTowardDirection(segmentId, nodeId, ArrowDirection.Right);
-                int forwardLanesCount = CountTargetLanesTowardDirection(segmentId, nodeId, ArrowDirection.Forward);
+                int leftLanesCount = LaneArrowManager.CountTargetLanesTowardDirection(segmentId, nodeId, ArrowDirection.Left);
```

Comment:
Aren't these 3 lines of code going to repeat the iteration in `CountTargetLanesTowardDirection()` three times?

Would it be worth having new `TallyTargetLanesPerDirection()` method that iterates once and returns how many lanes are in each direction via `out` params?

Reply:
This is not part of the critical path of the code. And its not the bottleneck either. The time spent here pales in comparison to the time spent calling the managers to apply junction restrictions/priority rules/... which are surprisingly slow! Optimizing this will have no noticeable influence on the speed of the code. But if you feel strongly about this I can Optimize it.

Reply:
If there are bigger optimisation gains elsewhere, then we should focus on those.